addons/hr/static/tests/m2x_avatar_employee.test.js

```diff
@@ -0,0 +1,541 @@
+    await contains(queryFirst(".o_avatar_card_buttons button")).click();
+    expect(".o-mail-ChatWindow").toHaveCount(2);
+    expect(
+        queryAll(
+            ".o-mail-ChatWindow-header button.o-dropdown.o-mail-ChatWindow-command > .text-truncate"
+        )[0].textContent
+    ).toBe("Yoshi");
+});
+
+test("many2many_avatar_employee widget in kanban view", async () => {
+    const { env } = await makeMockServer();
+    const [partnerId_1, partnerId_2] = env["res.partner"].create([
+        { name: "Mario" },
+        { name: "Luigi" },
+    ]);
+    const [userId_1, userId_2] = env["res.users"].create([
+        { partner_id: partnerId_1 },
+        { partner_id: partnerId_2 },
+    ]);
+    const [employeeId_1, employeeId_2] = env["hr.employee.public"].create([
+        {
+            user_id: userId_1,
+            user_partner_id: partnerId_1,
+            name: "Mario",
+            work_email: "Mario@partner.com",
+        },
+        {
+            name: "Luigi",
+            user_id: userId_2,
+            user_partner_id: partnerId_2,
+        },
+    ]);
+    env["m2x.avatar.employee"].create({
+        employee_ids: [employeeId_1, employeeId_2],
+    });
+    onRpc("has_group", () => false);
+    await start();
+    await mountView({
+        type: "kanban",
+        resModel: "m2x.avatar.employee",
+        arch: `<kanban>
+            <templates>
+                <t t-name="card">
+                    <footer>
+                        <field name="employee_ids" widget="many2many_avatar_employee"/>
+                    </footer>
+                </t>
+            </templates>
+        </kanban>`,
+    });
+    expect(
+        ".o_kanban_record:first .o_field_many2many_avatar_employee img.o_m2m_avatar"
+    ).toHaveCount(2);
+    expect(
+        queryAllAttributes(
+            ".o_kanban_record .o_field_many2many_avatar_employee img.o_m2m_avatar",
+            "data-src"
+        )[0]
+    ).toBe(`${getOrigin()}/web/image/hr.employee.public/${employeeId_2}/avatar_128`);
+    expect(
+        queryAllAttributes(
+            ".o_kanban_record .o_field_many2many_avatar_employee img.o_m2m_avatar",
+            "data-src"
+        )[1]
+    ).toBe(`${getOrigin()}/web/image/hr.employee.public/${employeeId_1}/avatar_128`);
+
+    // Clicking on first employee's avatar
+    await contains(queryAll(".o_kanban_record img.o_m2m_avatar")[1]).click();
+    await contains(".o_avatar_card");
+    expect(".o_card_user_infos > span").toHaveText("Mario");
+    expect(".o_card_user_infos > a").toHaveText("Mario@partner.com");
+    expect(queryFirst(".o_avatar_card_buttons button").textContent).toBe("Send message");
+    await contains(queryFirst(".o_avatar_card_buttons button")).click();
+    await contains(".o-mail-ChatWindow");
+    expect(
+        queryFirst(
+            ".o-mail-ChatWindow-header button.o-dropdown.o-mail-ChatWindow-command > .text-truncate"
+        ).textContent
+    ).toBe("Mario");
```

Comment:
Why not just `await waitFor(".o-mail-ChatWindow-header:contains('Mario')")`? Much shorter so easier to see.

Avoid long selectors, these are hard to maintain. `>` is a big no no: this makes the test heavily dependent on a given layout, which makes the test fail whenever the template of chat window changes. We don't like test that enforce an heavily based implementation. If you expect us to maintain this test every time we change it: no this sucks, I would be angry and just remove it rather than maintain it :D

This commit applies to whole file.